usage-rules.md

```diff
@@ -205,6 +205,31 @@ action :analyze_sentiment, :atom do
 end
 ```
 
+### Dynamic LLM Configuration
+
+For runtime configuration (like environment variables), use a function to define the LLM:
+
+```elixir
+action :analyze_sentiment, :atom do
+  argument :text, :string, allow_nil?: false
+  
+  run prompt(
+    fn _input, _context -> 
+      LangChain.ChatModels.ChatOpenAI.new!(%{
+        model: "gpt-4o",
+        api_key: System.get_env("OPENAI_API_KEY"),
```

Comment:
```suggestion
        # this can also be configured in application config, see langchain docs for more.
        api_key: System.get_env("OPENAI_API_KEY"),
```